### ISSUE TITLE
🐛 Bug Report: Code 400, URL is required, but createMembership method its optional.

### ISSUE THREAD

**Issue by xuelink** [appwrite/appwrite#6839]
### 👟 Reproduction steps

This is the method which only requires `teamId` and `roles`, others are optional like `email?` and `url?` so on.

```typescript
(method) Teams.createMembership(teamId: string, roles: string[], email?: string, userId?: string, phone?: string, url?: string, name?: string): Promise<Models.Membership>
```

Before we start, let me show you where this bug comes from

1.3.x -> `url` is required due to [docs](https://appwrite.io/docs/references/1.3.x/client-web/teams#createMembership)
1.4.x -> `url` is NOT required due to [docs](https://appwrite.io/docs/references/1.4.x/client-web/teams#createMembership)

1- Create a Team, or use any team for reproduction
```typescript
    this.api.teams.create(ID.unique(), name, ['owner']);
```
2-Try to createMembership
```typescript
    this.api.teams.createMembership(teamId, ['owner'], 'user@example.com');

    // OR
    this.api.teams.createMembership(teamId, ['owner'], 'user@example.com', userId);

    // OR
    this.api.teams.createMembership(teamId, ['owner'], undefined, userId);
```

The same error as a response:

<img width="413" alt="Screenshot 2023-10-07 at 11 24 28" src="https://github.com/appwrite/appwrite/assets/3280475/c8b69043-23b3-4290-b0d8-e2988d13be46">

### 👍 Expected behavior

It may give an error related what is exactly wrong with the request!

### 👎 Actual Behavior

Only its returning 404, URL is required.

### 💭 Extra Question/Suggestion for Documentary

In [documentation](https://appwrite.io/docs/references/1.4.x/client-web/teams#createMembership), 

> You only need to provide one of a user ID, email, or phone number. Appwrite will prioritize accepting the user ID > email > phone number if you provide more than one of these parameters.

But the method accepts first `email` then `userId`
Confusing, why don't you reorder the `createMembership` method inputs like in the docs or vice versa?

### 🎲 Appwrite version

Appwrite Version 1.4.3
[appwrite/sdk-for-web](https://github.com/appwrite/sdk-for-web): 13.0.0

### 💻 Operating system

Linux

### 🧱 Your Environment

```sh
root@appwrite:~/appwrite# docker compose exec appwrite doctor
  __   ____  ____  _  _  ____  __  ____  ____     __  __
 / _\ (  _ \(  _ \/ )( \(  _ \(  )(_  _)(  __)   (  )/  \
/    \ ) __/ ) __/\ /\ / )   / )(   )(   ) _)  _  )((  O )
\_/\_/(__)  (__)  (_/\_)(__\_)(__) (__) (____)(_)(__)\__/

👩‍⚕️ Running Appwrite Doctor for version 1.4.3 ...

[Settings]
🟢 Hostname has a public suffix (db.example.com)
🟢 CNAME target has a public suffix (db.example.com)
🟢 Using a unique secret key for encryption
🟢 App environment is set for production
🔴 Abuse protection is disabled
🟢 Console access limits are enabled
🟢 HTTPS force option is enabled
🔴 Logging adapter is disabled

[Connectivity]
🟢 Console.DB(console)...............................connected
🟢 Projects.DB(database_db_main).....................connected
🟢 Cache(cache_redis_main)...........................connected
🟢 Queue(queue)......................................connected
🟢 PubSub(pubsub)....................................connected
🟢 SMTP..............................................connected

[Volumes]
🟢 Uploads Volume is readable
🟢 Uploads Volume is writeable
🟢 Cache Volume is readable
🟢 Cache Volume is writeable
🟢 Config Volume is readable
🟢 Config Volume is writeable
🟢 Certs Volume is readable
🟢 Certs Volume is writeable

[Disk Space]
🟢 Uploads Volume has 43.41GB free space (16.26% used)
🟢 Cache Volume has 43.41GB free space (16.26% used)
🟢 Config Volume has 43.41GB free space (16.26% used)
🟢 Certs Volume has 43.41GB free space (16.26% used)

Failed to check for a newer version
```

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by behicsakar** [comment]
I tried to bypass this bug with giving a dummy `url` as an input

```typescript
    this.api.teams.createMembership(teamId, ['owner'], undefined, userId, undefined, 'http://localhost');
```

It created membership but it is also buggy, in user membership its not shown but when the team memberships you can see 2 members as shown in next kap.

https://github.com/appwrite/appwrite/assets/3280475/ed123980-f1c9-4392-a5ff-c0baec660fdc

**Comment by stnguyen90** [comment]
@behicsakar thanks for creating this issue! We'll look into this soon.

> But the method accepts first email then userId
Confusing, why don't you reorder the createMembership method inputs like in the docs or vice versa?

There was a bug with the docs but a fix should be released soon.

**Comment by stnguyen90** [comment]
> 1.4.x -> url is NOT required due to [docs](https://appwrite.io/docs/references/1.4.x/client-web/teams#createMembership)

@behicsakar, it looks like you're using the web SDK. The URL param is required when using a client SDK. We need to update the docs to clarify that.

**Comment by behicsakar** [comment]
> > 1.4.x -> url is NOT required due to [docs](https://appwrite.io/docs/references/1.4.x/client-web/teams#createMembership)
> 
> @behicsakar, it looks like you're using the web SDK. The URL param is required when using a client SDK. We need to update the docs to clarify that.

It will be better to clarify, confusing like that! It sounds like a sdk bug but its not as you say

**Comment by nick2432** [comment]
can i work on this issue?

**Comment by stnguyen90** [comment]
@nick2432, are you still interested in working on this?

@gewenyu99, should we update the description or the url param? What should the copy be?

**Comment by stnguyen90** [comment]
Closing as a duplicate of https://github.com/appwrite/website/issues/635
